simulator/src/main/java/com/github/benmanes/caffeine/cache/simulator/parser/rewrite/OutputFormat.java

```diff
@@ -18,18 +18,27 @@
 import java.io.BufferedWriter;
 import java.io.IOException;
 
+import com.github.benmanes.caffeine.cache.simulator.policy.AccessEvent;
+
 /**
  * The trace output format.
  *
  * @author ben.manes@gmail.com (Ben Manes)
  */
 public enum OutputFormat {
   CLIMB {
-    @Override public void write(BufferedWriter writer, long event) throws IOException {
-      writer.write(Long.toString(event));
+    @Override public void write(BufferedWriter writer, AccessEvent event) throws IOException {
+      writer.write(Long.toString(event.key()));
+      writer.newLine();
+    }
+  },
+  ADAPT_SIZE {
+    long time = 0;
```

Comment:
This is a fair pain, because you want the state but it is implicitly static which is quite ugly to modify. Adding it as a parameter would be preferable.

Reply:
What do you mean by "as a parameter"? Since AccessEvent does not have a time field the time can only be synthetic.

Reply:
To the `OutputFormat.write` method and keep the state in the `Rewriter`.  An enum should be stateless as it represents a constant and implicitly static.